pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -107,11 +107,46 @@ OSD_ID="$ROOK_OSD_ID"
 OSD_UUID=%s
 OSD_STORE_FLAG="%s"
 OSD_DATA_DIR=/var/lib/ceph/osd/ceph-"$OSD_ID"
+KEYRING_FILE="$OSD_DATA_DIR"/keyring
 CV_MODE=%s
 DEVICE="$%s"
 
-# create new keyring
-ceph -n client.admin auth get-or-create osd."$OSD_ID" mon 'allow profile osd' mgr 'allow profile osd' osd 'allow *' -k /etc/ceph/admin-keyring-store/keyring
+# In rare cases keyring file created with prepare-osd but did not
+# being stored in ceph auth system therefore we need to import it
```

Comment:
Do you know how this race happens in code level? If possible, I'd like to fix the root cause (race) than bypassing it.

In addition, have you ever verified your problem was fixed in this PR? IMO, it's not necessary to add a new test in this PR. However, we need to confirm that this PR really resolve your problme at least the merge time.

Reply:
> Do you know how this race happens in code level? If possible, I'd like to fix the root cause (race) than bypassing it.

I don't know why provision container misses adding keyring to ceph cluster because all we got is "Creating keyring file" log line which implies creating keyring on ceph side and creating keyring file on host. I tried to understand where the issue could be during keyring creation but did not reach any result.

> In addition, have you ever verified your problem was fixed in thisIn addition, have you ever verified your problem was fixed in this

Yes, we verified that.